github/downloader_test.go

```diff
@@ -80,74 +123,380 @@ func getDownloader() (*Downloader, *testutils.Memory, error) {
+	}
+
+}
 
+func checkIssuePRComments(require *require.Assertions, db *sql.DB, oracle testutils.RepositoryTest, strict bool) {
+	var numOfComments int
+	err := db.QueryRow("select count(*) from issue_comments where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfComments)
+	require.NoError(err, "Error in retrieving issue comments")
+	// NB: ghsync saves both Issue and PRs comments in the same table, issue_comments => See store/db.go comment
+	if strict {
+		require.Equal(oracle.NumOfPRComments+oracle.NumOfIssueComments, numOfComments, "Issue and PR Comments")
+	} else {
+		require.GreaterOrEqual(oracle.NumOfPRComments+oracle.NumOfIssueComments, numOfComments, "Issue and PR Comments")
+	}
+}
+
+func checkPRs(require *require.Assertions, db *sql.DB, oracle testutils.RepositoryTest, strict bool) {
+	var numOfPRs int
+	err := db.QueryRow("select count(*) from pull_requests where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfPRs)
+	require.NoError(err, "Error in retrieving pull requests")
+	if strict {
+		require.Equal(oracle.NumOfPRs, numOfPRs, "PRs")
+	} else {
+		require.GreaterOrEqual(oracle.NumOfPRs, numOfPRs, "PRs")
+	}
+}
+
+func checkPRReviewComments(require *require.Assertions, db *sql.DB, oracle testutils.RepositoryTest, strict bool) {
+	var numOfPRReviewComments int
+	err := db.QueryRow("select count(*) from pull_request_comments where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfPRReviewComments)
+	require.NoError(err, "Error in retrieving pull request comments")
+	if strict {
+		require.Equal(oracle.NumOfPRReviewComments, numOfPRReviewComments, "PR Review Comments")
+	} else {
+		require.GreaterOrEqual(oracle.NumOfPRReviewComments, numOfPRReviewComments, "PR Review Comments")
+	}
+}
+
+func checkRepo(require *require.Assertions, db *sql.DB, oracle testutils.RepositoryTest, strict bool) {
+	var (
+		htmlurl   string
+		createdAt time.Time
+		private   bool
+		archived  bool
+		hasWiki   bool
+		topics    []string
+	)
+	err := db.QueryRow("select htmlurl, created_at, private, archived, has_wiki, topics from repositories where owner_login = $1 and name = $2", oracle.Owner, oracle.Repository).Scan(&htmlurl, &createdAt, &private, &archived, &hasWiki, pq.Array(&topics))
+	require.NoError(err, "Error in retrieving repo")
+	require.Equal(oracle.URL, htmlurl)
+	require.Equal(oracle.CreatedAt, createdAt.String())
+	require.Equal(oracle.IsPrivate, private)
+	require.Equal(oracle.IsArchived, archived)
+	require.Equal(oracle.HasWiki, hasWiki)
+	require.ElementsMatch(oracle.Topics, topics)
+}
+
+// TestOnlineRepositoryDownloadWithDB Tests the download of known and fixed GitHub organization and stores it in a Postgresql DB
+func (suite *DownloaderTestSuite) TestOnlineRepositoryDownloadWithDB() {
+	t := suite.T()
+	checkToken(t)
+	tests, err := loadTests("../testdata/online-repository-tests.json")
```

Comment:
if other files inside `../testdata/` are constants, maybe it will make sense to move this one to constant also?